### PR TITLE
Feature/3 authenticate by basic auth

### LINKED ISSUE #3: Implement basic authentication as alternative auth method
API token is okay but sometimes basic authentication with `username/password` must do as well. Currently, Basich Auth is not supported but should be. Getting an API token is cool but too often it takes away time in rapidly deployed environments.

### PR DESCRIPTION
**Pull request by ppxl**
This PR adds basic authentication (to the entities below) and SSL cert skipping and thusly resolves #3 and #4.

Entities that optionally receive basic auth when this PR is merged:
- projects (CRUD)
- issues (CRUD)
- issue categories (CRUD)
- versions (CRUD)
- issue priorities (only Read, there is no other implementation)

Further changes:

`go-redmine`'s constructor is doomed to be defunct because future developments would expand the parameter list only further and further. The previous change (introducing APIAuth) manifested a rather awkward usage of the constructor. Thus, this PR deprecates the original client constructor in favour of a way more flexible `ClientBuilder`. This ClientBuilder also eases the pain of providing an APIAuth object by easy builder methods for Basic Auth or API Token auth.

This PR adds also proper Jenkins CI with unit tests, code coverage, and SonarQube scans. The CLI-package is excluded from the SonarQube scanning because it is most likely that it never will be used (or deleted even). The API package is where the beef is at.

The usage of the new features was documented, and previous documentation moves to the centralized docs/ directory as well.

### CODE REVIEW

File: cmd/godmine/main.go [hunk 1]
```diff
@@ -238,11 +240,14 @@ func getConfig() config {
 
 func addIssue(subject, description string) {
 	var issue redmine.Issue
-	c := redmine.NewClient(conf.Endpoint, conf.Apikey)
+	c, err := redmine.NewClient(conf.Endpoint, auth)
```
Comment (cesmarvin) [inline cmd/godmine/main.go]: 🚫 **[checkstyle]** <staticcheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>SA1019: redmine.NewClient is deprecated: Use redmine.ClientBuilder to create a redmine client that supports more options and detects configuration mistakes.

File: cmd/godmine/main.go [hunk 2]
```diff
@@ -253,7 +258,10 @@ func createIssue() {
 	if err != nil {
 		fatal("%s\n", err)
 	}
-	c := redmine.NewClient(conf.Endpoint, conf.Apikey)
+	c, err := redmine.NewClient(conf.Endpoint, auth)
```
Comment (cesmarvin) [inline cmd/godmine/main.go]: 🚫 **[checkstyle]** <staticcheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>SA1019: redmine.NewClient is deprecated: Use redmine.ClientBuilder to create a redmine client that supports more options and detects configuration mistakes.

File: cmd/godmine/main.go [hunk 3]
```diff
@@ -262,7 +270,10 @@ func createIssue() {
 }
 
 func updateIssue(id int) {
-	c := redmine.NewClient(conf.Endpoint, conf.Apikey)
+	c, err := redmine.NewClient(conf.Endpoint, auth)
```
Comment (cesmarvin) [inline cmd/godmine/main.go]: 🚫 **[checkstyle]** <staticcheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>SA1019: redmine.NewClient is deprecated: Use redmine.ClientBuilder to create a redmine client that supports more options and detects configuration mistakes.

File: client.go [hunk 4]
```diff
@@ -10,40 +13,165 @@ import (
...
+	AuthTypeTokenQueryParam
+	AuthTypeBasicAuthWithTokenPassword
+	AuthTypeNoAuth
+)
+
+const (
+	httpMethodGet    = "GET"
+	httpMethodPost   = "POST"
+	httpMethodPut    = "PUT"
+	httpMethodDelete = "DELETE"
```
Comment (cbeyer42) [inline client.go]: Why are you defining new constants for the http methods? You can use predefined constants from the `http` package e. g. `http.MethodPut`.

Comment (ppxl) [reply client.go]: Done with commit 939fb234f5541b20bacbb56ea6bdb9efcd75b29d

**Pull request merged**